### ISSUE TITLE
(feature request) Highlight matching quotes, double quotes and backticks

### ISSUE THREAD

**Issue by darkred** [Microsoft/vscode#15899]
- VSCode Version: 1.7.1
- OS Version: Win10 x64

Steps to Reproduce:

My suggestion is to not only highlight matching square brackets, round brackets/parentheses and curly brackets (`[], (), {}`) -currently- as well as matching tags e.g. `<div></div>`,
**but also quotes, double quotes and backticks,** i.e. `''`, `""` and ` `` `.
_The highlighting would be via **underlining** these symbols,
in order to visually distinguish them from any matching brackets (which are highlighted with outline (box))._

This is a feature of one of the most popular packages for Sublime Text, BracketHighlighter. 

<br>

PS. The [template literals (which are quoted via **backticks**)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are an ES6 feature.

**Comment by mattflix** [comment]
+1

But, why underlining for certain matches (various quoting marks).  Why not use the same box in all cases?

**Comment by darkred** [comment]
VSCode uses outline(box) to highlight the matching brackets.
Sublime Text highlights them with underline. Brackethighliter also uses underline (+gutter marks) for highlighting the matching quotes.

Brackets are used (at least in JavaScript) to enclose: code blocks, objects, functions, arrays, 
and for accessing properties (in bracket notation) in btw which also uses quotes for property names.
_This is the "classic" highlighting which all editors offer._

Quotes, on the other hand, are only used for strings or quoted strings inside strings, and for object properties.
Highlighting quotes is something extra to the "classic" highlighting, 
that's why I suggest it to differ.
That's my humble opinion, anyway.

**Comment by Christo44** [comment]
It will probably happen, but it would be great if this can be implemented for tags too

**Comment by cbrnr** [comment]
Atom lets you configure the autocomplete characters, and all kinds of quotes and ticks are enabled by default.

![screen shot 2017-11-14 at 22 26 26](https://user-images.githubusercontent.com/4377312/32805844-f273fade-c98a-11e7-951c-30e7d1ddecb0.png)

The "Wrap selection in brackets" is especially handy when it also works for backticks, e.g. when writing Markdown documents.

**Comment by jasonkuhrt** [comment]
Great for jest snapshots of very large strings.

**Comment by mkanet** [comment]
Does anyone know of an extension that does this?  It's one feature I really miss that I used to have in Sublime Text via the Bracket Highlighter addon.

**Comment by kanlukasz** [comment]
> 
> 
> Does anyone know of an extension that does this? It's one feature I really miss that I used to have in Sublime Text via the Bracket Highlighter addon.

You can try https://marketplace.visualstudio.com/items?itemName=rafamel.subtle-brackets

**Comment by mkanet** [comment]
@kanlukasz, I don't think that extension supports single and double quote highlighting.  It didn't seem to like it when I specified single/double quote characters for its settings.

```
"subtleBarckets.pairs" : [
  {
    "open": """,
    "close": """
  },
 {
    "open": "'",
    "close": "'"
  }
]
```
